File: src/main/java/client/inventory/item/templates/MobSummonItemTemplate.java
package client.inventory.item.templates;

import util.packet.PacketReader;

import java.util.ArrayList;
import java.util.List;

public class MobSummonItemTemplate extends ItemBundleTemplate {

    private final List<MobSummonItemEntry> mobs = new ArrayList<>();

    public List<MobSummonItemEntry> getMobs() {
        return mobs;
    }

    public MobSummonItemTemplate(int id, PacketReader r) {
        super(id, r);
        int size = r.readShort();
        for (int i = 0; i < size; i++) {
            mobs.add(new MobSummonItemEntry(r));
        }
    }

    public static class MobSummonItemEntry {

        private final int templateId, prob;

        public int getTemplateId() {
            return templateId;
        }

        public int getProb() {
            return prob;
        }

        public MobSummonItemEntry(PacketReader r) {
            this.templateId = r.readInteger();
            this.prob = r.readInteger();
        }
    }
}

File: src/main/java/client/inventory/item/templates/MobSummonItemTemplate.kt
package client.inventory.item.templates

import util.packet.PacketReader
import java.util.*

class MobSummonItemTemplate(id: Int) : ItemBundleTemplate(id) {

    private val mobs: MutableList<MobSummonItemEntry> = ArrayList()

    override fun decode(r: PacketReader): MobSummonItemTemplate {
        super.decode(r)

        val size = r.readShort().toInt()
        for (i in 0 until size) {
            mobs.add(MobSummonItemEntry(r))
        }

        return this
    }

    class MobSummonItemEntry(r: PacketReader) {
        val templateId: Int
        val prob: Int

        init {
            templateId = r.readInteger()
            prob = r.readInteger()
        }
    }
}